UI/SidePanel.cpp

```diff
@@ -1835,7 +1840,66 @@ void SidePanel::PlanetPanel::Refresh() {
     if (m_specials_panel)
         m_specials_panel->Update();
 
-    // set stealth browse text
+    // create planet status marker
+    if (planet->OwnedBy(client_empire_id)) {
+        DetachChild(m_planet_status_btn);
+        std::vector<std::string> planet_status_messages;
+        std::shared_ptr<GG::Texture> planet_status_texture;
+
+        // status: no supply
+        if (!GetSupplyManager().SystemHasFleetSupply(planet->SystemID(), planet->Owner())) {
+            planet_status_messages.emplace_back(boost::io::str(FlexibleFormat(
+                                                UserString("OPTIONS_DB_UI_PLANET_STATUS_NO_SUPPLY")) % planet->Name()));
+            planet_status_texture = ClientUI::GetTexture(ClientUI::ArtDir() / "icons" / "planet_status_supply.png", true);
+        }
+
+        // status: attacked on previous turn
+        if (planet->LastTurnAttackedByShip() == CurrentTurn() - 1) {
+            planet_status_messages.emplace_back(boost::io::str(FlexibleFormat(
+                                                UserString("OPTIONS_DB_UI_PLANET_STATUS_ATTACKED")) % planet->Name()));
+            planet_status_texture = ClientUI::GetTexture(ClientUI::ArtDir() / "icons" / "planet_status_attacked.png", true);
+        }
+
+        // status: conquered on previous turn
+        if (planet->LastTurnConquered() == CurrentTurn() - 1) {
+            planet_status_messages.emplace_back(boost::io::str(FlexibleFormat(
+                                                UserString("OPTIONS_DB_UI_PLANET_STATUS_CONQUERED")) % planet->Name()));
+            planet_status_texture = ClientUI::GetTexture(ClientUI::ArtDir() / "icons" / "planet_status_conquered.png", true);
+        }
+
+        // status: very unhappy
+        if ((planet->GetMeter(METER_HAPPINESS)->Current() <= 5) && (planet->GetMeter(METER_POPULATION)->Current() > 0)) {
+            planet_status_messages.emplace_back(boost::io::str(FlexibleFormat(
+                UserString("OPTIONS_DB_UI_PLANET_STATUS_UNHAPPY")) % planet->Name()));
+            planet_status_texture = ClientUI::GetTexture(ClientUI::ArtDir() / "icons" / "planet_status_unhappy.png", true);
+        }
+
+        // status: bombarded (TBD)
+
+        // status: several
+        if (planet_status_messages.size() > 1) {
```

Comment:
Any way to arrange multiple buttons to be shown simultaneously instead of switching to a generic icon in that case?